### ISSUE TITLE
'kubectl help' off-by-one error parsing command line

### ISSUE THREAD

**Issue by mml** [kubernetes/kubernetes#28707]
```console
% _output/dockerized/bin/linux/amd64/kubectl help run       


Available Sub-commands:
  get            Display one or many resources
  set            Set specific features on objects
  describe       Show details of a specific resource or group of resources
  create         Create a resource by filename or stdin
[...]

% _output/dockerized/bin/linux/amd64/kubectl help run run
Create and run a particular image, possibly replicated.
Creates a deployment or job to manage the created container(s).

Aliases:
  run, run-container

Examples:

# Start a single instance of nginx.
kubectl run nginx --image=nginx

# Start a single instance of hazelcast and let the container expose port 5701 .
kubectl run hazelcast --image=hazelcast --port=5701
[...]

% _output/dockerized/bin/linux/amd64/kubectl version     
Client Version: version.Info{Major:"1", Minor:"4+", GitVersion:"v1.4.0-alpha.0.1439+04602bb9e52a51-dirty", GitCommit:"04602bb9e52a511e422616f36a126b8b92b74afd", GitTreeState:"dirty", BuildDate:"2016-07-08T20:00:27Z", GoVersion:"go1.6.2", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"4+", GitVersion:"v1.4.0-alpha.0.1273+d618e5329bfb86-dirty", GitCommit:"d618e5329bfb86c266ab324add9bcf10cdb77bbc", GitTreeState:"dirty", BuildDate:"2016-07-06T22:59:24Z", GoVersion:"go1.6.2", Compiler:"gc", Platform:"linux/amd64"}
```

**Comment by lojies** [comment]
this has been fixed in
[https://github.com/kubernetes/kubernetes/commit/b2f8ee4f122a3130b10f2f1bea64d96361f6cf81]

**Comment by asalkeld** [comment]
Please close the issue, PR has been merged.

**Comment by mml** [comment]
Fixed in #29737
